compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1723,6 +1723,7 @@ rustc_queries! {
         desc { |tcx| "computing visibility of `{}`", tcx.def_path_str(def_id) }
         separate_provide_extern
         feedable
+        cache_on_disk_if { def_id.is_local() }
```

Comment:
Should this be benchmarked and landed separately?

Reply:
Oh whoops, this is not a perf thing, this was a hack to avoid having to implement

> need to make feeding queries a side effect, too. At least ones that aren't written to disk

Reply:
Ok, I looked into how we could design this, and I think the best way to do this is to require queries that get fed to also get cached to disk.


The only other alternative I came up with is to store all fed queries in the side effect table of their caller (somewhat hard, because we need to store all possible key/value combinations)
This also feels like it duplicates the cache-to-disk logic, and definitely does if the query is also marked as cache-to-disk.

Reply:
> I think the best way to do this is to require queries that get fed to also get cached to disk.

I no longer think this. That would require thinking about HIR in incremental, at minimum due to the `opt_hir_owner_nodes` query, but `local_def_id_to_hir_id` has no real way of being cached